infra-parent/infra-v2/src/main/java/com/redhat/service/smartevents/infra/v2/api/models/ConditionStatus.java

```diff
@@ -8,7 +8,8 @@ public enum ConditionStatus implements BaseEnumeration {
     TRUE("True"),
     FALSE("False"),
     UNKNOWN("Unknown"),
-    FAILED("Failed");
+    FAILED("Failed"),
+    DELETED("Deleted");
```

Comment:
We don't have such logic for a `DELETED` status in the CP. In case the resources are deleted, there should be a condition like `ResourceDeleted` with status `TRUE`.

Reply:
Agreed with @r00ta . IIRC the proposal was to have a `Condition` in the form `Deleted=true` once the resource has been deleted.

Reply:
I prefer `XXXDeleted=Unknown|True|False|Failed` too.

It is symmetrical with the _creation_ `Condition`'s of `XXXReady=Unknown|True|False|Failed`.

Reply:
done.. I create a new Condition with type `BridgeDeleted` and set it to `True`

Reply:
Nooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo.

Your whole logic remains a bit off. The `manager` wants to know as each dependency is removed; much the same as it was advised as they were created. The `manager` tracks "delete requested" and "deleting" before the final deletion. "delete request" is when the User has requested it. "deleting" is when activities are in progress for the deletion (e.g. delete DNS entry, delete Isitio configuration, delete knative broker etc - whatever) and "deleted" is the final update needed to signal the `manager` to actually delete database records.

We need conditions symmetrical to _create_ for _delete_. For example:-
```
    public static final String DP_SECRET_DELETED_NAME = "SecretDeleted";
    public static final String DP_CONFIG_MAP_DELETED_NAME = "ConfigMapDeleted";
    public static final String DP_KNATIVE_BROKER_DELETED_NAME = "KNativeBrokerDeleted";
    public static final String DP_AUTHORISATION_POLICY_DELETED_NAME = "AuthorisationPolicyDeleted";
    public static final String DP_NETWORK_RESOURCE_DELETED_NAME = "NetworkResourceDeleted";
```
IDK if they are all needed (as k8s will delete some automagically) but for the resources we need to delete manually I'd have thought it preferential to have an explicit `Condition` for them (so that we'd know what had been deleted successfully if the `operator` pod died before deletion had fully completed).

I guess you don't handle deletion [yet](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/468203f50e91160441a47cda31801438e84d7787/shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/ManagedBridgeServiceImpl.java#L123).